src/service_daemon.rs

```diff
@@ -3851,13 +3856,10 @@ mod tests {
         let mut resolved = false;
 
         while let Ok(event) = browse_chan.recv_timeout(timeout) {
-            match event {
```

Comment:
Those next 3 changes were suggested by rust clippy for [single_match](https://rust-lang.github.io/rust-clippy/master/index.html#single_match), if not feasible please let me know.

Reply:
I've reverted all unrelated changes, to avoid adding too much refactoring to this PR